__tests__/CognitoUtils.test.js

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 Stanford University see Apache2.txt for license
+
+import { AuthenticationDetails, CognitoUser, CognitoUserPool } from 'amazon-cognito-identity-js'
+import CognitoUtils from '../src/CognitoUtils'
+import Config from '../src/Config'
+
+describe('CognitoUtils', () => {
+  const username = 'testuser'
+  const password = 'testpass'
+
+  describe('authenticationDetails()', () => {
+    it('returns an instance of AuthenticationDetails with passed credentials', () => {
+      const details = CognitoUtils.authenticationDetails(username, password)
+      expect(details).toBeInstanceOf(AuthenticationDetails)
+      expect(details.username).toEqual(username)
+      expect(details.password).toEqual(password)
+    })
+  })
+
+  describe('cognitoUserPool()', () => {
+    it('returns an instance of CognitoUserPool with configured user pool and client IDs', () => {
+      const pool = CognitoUtils.cognitoUserPool()
+      expect(pool).toBeInstanceOf(CognitoUserPool)
+      expect(pool.userPoolId).toEqual(Config.awsCognitoUserPoolId)
+      expect(pool.clientId).toEqual(Config.awsClientID)
+    })
+  })
+
+  describe('cognitoUser()', () => {
+    it('returns an instance of CognitoUser with username and user pool', () => {
+      const user = CognitoUtils.cognitoUser(username)
+      expect(user).toBeInstanceOf(CognitoUser)
+      expect(user.username).toEqual(username)
+      expect(user.pool).toBeInstanceOf(CognitoUserPool)
```

Comment:
i would be slightly inclined to check that the pool has the right details... but that might be overkill?